Fix failing AndroidTests

<!--
If you've never submitted an issue to the SORMAS repository before or this is your first time using this template, please read the Contributing guidelines (https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/docs/CONTRIBUTING.md) for an explanation of the information we need you to provide. You don't have to remove this comment or any other comment from this issue as they will automatically be hidden.
-->
### Bug Description
A number of android tests fail
![androidTests2](https://user-images.githubusercontent.com/5616564/133510408-e03b0d1f-231c-4d23-a159-583ddef76ccb.png)
![androidTests1](https://user-images.githubusercontent.com/5616564/133510413-d7f694a5-c8c9-4eea-a33c-23fd806e9418.png)

Some of the tests do not reflect the current state of the implementation:
- `CaseBackendTest..shouldUpdateCaseAndAssociatedEntitiesOnMove()` manipulates `Case.district`. However, task reassignment is triggered by a change of `Case.responsibleDistrict` or - if `responsibleDistrict` is null, which is not the case here - by a change of `district`. Thus, neither assignment of a new `surveillanceOfficer` nor a reassignment of tasks occur in the course of the test. Equallly, hospitalizations are not updated as expected in the test.

Some of the tests fail depending of other tests run before:
- `CaseBackendTest.shouldCreateCase()` succeeds in the first run with no tests run before, but fails in a second run.

### Steps to Reproduce
<!-- Optional; please add more steps if necessary -->
Run android tests in android studio, either one by one or all together

### Expected Behavior
Android tests should succeed, test result should be independent of other tests run  before.
Database and other persistent states should be cleaned and prepared before each test.

### Screenshots
<!-- Optional -->

### System Details
<!-- Mandatory; you only have to specify the Server URL if the error appeared on a publicly available test server -->
* Device:
* SORMAS version: 1.64.0-SNAPSHOT
* Android version/Browser:
* Server URL:
* User Role:

### Additional Information
<!-- Optional -->

Running the android tests from Android Studio one by one work, but when running them in bulk only the first one works and all the others
```
Caused by: android.database.sqlite.SQLiteReadOnlyDatabaseException: attempt to write a readonly database (code 1032)
at android.database.sqlite.SQLiteConnection.nativeExecute(Native Method)
at android.database.sqlite.SQLiteConnection.execute(SQLiteConnection.java:555)
at android.database.sqlite.SQLiteSession.execute(SQLiteSession.java:619)
at android.database.sqlite.SQLiteStatement.execute(SQLiteStatement.java:44)
at com.j256.ormlite.android.AndroidDatabaseConnection.update(AndroidDatabaseConnection.java:316)
... 38 more
```

Changing the `initTest` method of a class to be ran `@BeforeClass` does fix some issues, but introduces others.

Shouldn't this have been addressed with this issue?

--- a/sormas-app/app/src/androidTest/java/de/symeda/sormas/app/TestHelper.java
+++ b/sormas-app/app/src/androidTest/java/de/symeda/sormas/app/TestHelper.java
@@ -56,12 +56,13 @@ public class TestHelper {
 	public static void initTestEnvironment(boolean setInformantAsActiveUser) {
 		// Initialize a testing context to not operate on the actual database
 		Context context = InstrumentationRegistry.getInstrumentation().getTargetContext();
-		// Make sure that no database/user is still set from the last run
-		context.deleteDatabase(TEST_DATABASE_NAME);
 
-		ConfigProvider.clearUserLogin();
 		// Initialize the testing database
 		DatabaseHelper.init(context, TEST_DATABASE_NAME);
+		// Make sure that no database/user is still set from the last run
+		DatabaseHelper.clearTables(true);
+		DatabaseHelper.clearConfigTable();
+		ConfigProvider.clearUserLogin();
 		ConfigProvider.init(context);
 
 		ConfigProvider.setServerRestUrl("http://this-is-a-test-url-that-hopefully-doesnt-exist.com");

--- a/sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java
+++ b/sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java
@@ -291,6 +291,15 @@ public class DatabaseHelper extends OrmLiteSqliteOpenHelper {
 		}
 	}
 
+	public static void clearConfigTable() {
+		try {
+			TableUtils.clearTable(DatabaseHelper.getCaseDao().getConnectionSource(), Config.class);
+		} catch (SQLException e) {
+			Log.e(DatabaseHelper.class.getName(), "Can't clear config table", e);
+			throw new RuntimeException(e);
+		}
+	}
+
 	/**
 	 * This is called when the database is first created. Usually you should call createTable statements here to build
 	 * the tables that will store your data.

